linode/manager
# upcoming: [M3-7898] - Support ticket severity

## Description 📝
Adds Cloud Manager support for the upcoming support ticket severity changes.

## Changes  🔄
- APIv4:
  - Updated `SupportTicket` and `TicketRequest` types with new `severity` field
  - New `Support Ticket Severity` account capability
- Cloud Manager:
  - New "Severity" column in support ticket list
  - Display ticket severity in ticket details status bar
  - Add severity select input when creating a new support ticket

## Target release date 🗓️
4/15

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Severity column |
| ![Screenshot 2024-03-26 at 3 26 44 PM](https://github.com/linode/manager/assets/122488130/18b1890c-fe6d-418e-b003-fc11ef3e70b2) | ![Screenshot 2024-03-26 at 3 26 55 PM](https://github.com/linode/manager/assets/122488130/8752b953-8993-40e3-affd-917c1b13ab10) |
| Ticket details |
|  ![Screenshot 2024-03-26 at 3 28 25 PM](https://github.com/linode/manager/assets/122488130/b4ca80f6-be77-4b1c-8bf9-fc15126643cb) |  ![Screenshot 2024-03-26 at 3 28 30 PM](https://github.com/linode/manager/assets/122488130/efa00db5-4cd2-463e-8d4c-72fc7f0a7c21) |
| New Ticket Dialog |
| ![Screenshot 2024-03-26 at 3 28 58 PM](https://github.com/linode/manager/assets/122488130/4796cd23-516f-4926-b4f7-d81ee3ae6ccf) |  ![Screenshot 2024-03-26 at 3 29 15 PM](https://github.com/linode/manager/assets/122488130/c7e77b1e-7626-4924-863b-887cf2b88978) |

## How to test 🧪

### Prerequisites
- Enable 'Support Ticket Severity' feature flag
- Enable MSW

### Verification steps
- Navigate to Support > Tickets and verify that the 'Severity' column appears, displays each ticket's severity and can be sorted
- Click on a ticket and verify the ticket's severity appears in the status bar at the top of the page
- Click "Open New Ticket" and verify a severity can be selected
- Hover over tooltip and verify copy (from ticket)

> [!NOTE]
> As of 3/26, ticket severity is not supported by the API. Submitting a ticket will ignore the specified severity.

## As an Author I have considered 🤔

*Check all that apply*
